### ISSUE TITLE
Qiling can't open port although "ql.root = True" - "bind: Operation not permitted" 

### ISSUE THREAD

**Issue by RadCet** [qilingframework/qiling#931]
**Describe the bug**
- When trying to emulate httpd service, process try to bind to port 80 or 8080. 
- Already kill any process running at 80 or 8080 before
- But try to run with **sudo** and config qiling with "ql.root = True" still get "bind: Operation not permitted" 
- It happen with both 80 and 8080
- Qiling version 1.3.0
- Notice that when I try to run in a pi 3, everything is ok
 
**Sample Code**
```python
import sys
sys.path.append("..")
from qiling import *
from qiling.const import QL_VERBOSE

def run_sandbox(path, rootfs):
	ql = Qiling(path, rootfs, verbose=QL_VERBOSE.DEBUG)
	#ql.debugger = "127.0.0.1:9999"
	ql.root = True
	ql.run()


if __name__ == "__main__":
	run_sandbox(["rootfs/usr/sbin/httpd","-c","/etc/httpd.conf"], "rootfs")
```

**Screenshots**
- port 80
![image](https://user-images.githubusercontent.com/24670183/135818106-fff60b65-0967-4c07-92df-361caf6035ec.png)
- port 8080
![image](https://user-images.githubusercontent.com/24670183/135818348-2b915311-f1bc-4da5-acfc-33a510739a5f.png)

**Comment by elicn** [comment]
Thanks for reporting this, this is indeed a Qiling bug.

*TL;DR* - Debugging a few examples show that setting the `root` property after instantiating Qiling is rather too late.

*Details:*
Among some other things, Qiling init method sets up the OS instance, in which `os.uid` is set acording to the value appears in `ql.root` value (`False` by default). At the end of Qiling init, the loader [an ELF loader, in this case] runs and uses `os.uid` to set the correct value in the ELF auxiliary vector. It is just after that that the user may set `ql.root = True` in their code - which is already too late.

Since there is no effective way to propagate the `root` value after loader is done, I think the way to solve this would be adding an optional `root` argument to the Qiling init method, and make the existing `root` property read-only. Since there is already an argument named `rootfs`, perhaps it would be better to name the new argument `asroot` or `isroot` to prevent mixups, but perhaps `root` would be the clearest option after all..

@xwings, what do  you think? I can create a PR for this.

**Comment by xwings** [comment]
Maybe we can do it in profile? What do you think

**Comment by elicn** [comment]
> Maybe we can do it in profile? What do you think

Well, implicitly this is already the case: you can do that by setting both `uid` and `gid` on your profile to `0`. That would effectively make you root. However, I do think it is significant enough to set through object constructor [while consider removing others which can defenitly can be set through profile].
